File: 2015/nazionali/bottleneck/att/grader.pas
uses bottleneck;

const MAXN = 100000;
      MAXM = 1000000;
var N, M, W, L, R, C : longint;
    arco_da, arco_a, capacita : array[0..MAXM-1] of longint;
    i : longint;
    fr, fw : text;

begin
    assign(fr, 'input.txt');
    assign(fw, 'output.txt');
    reset(fr);
    rewrite(fw);

    readln(fr, N, M, W, L, R, C);
    for i:=0 to M-1 do
        readln(fr, arco_da[i], arco_a[i], capacita[i]);

    writeln(fw, Analizza(N, M, W, L, arco_da, arco_a, capacita, R, C));

    close(fr);
    close(fw);
end.

File: 2015/nazionali/bottleneck/att/grader.c
#include <stdio.h>
#include <assert.h>
#define MAXN 100000
#define MAXM 1000000

static FILE *fr, *fw;

static int N, M, W, L;
static int R, C;
static int arco_da[MAXM], arco_a[MAXM], capacita[MAXM];

int Analizza(int N, int M, int W, int L, int arco_da[], int arco_a[], int capacita[], int R, int C);

int main(){
    fr = fopen("input.txt", "r");
    fw = fopen("output.txt", "w");

    int i;
    assert(4 == fscanf(fr, "%d%d%d%d", &N, &M, &W, &L));
    assert(2 == fscanf(fr, "%d%d", &R, &C));
    for (i=0; i<M; i++)
        assert(3 == fscanf(fr, "%d%d%d", arco_da+i, arco_a+i, capacita+i));

    fprintf(fw, "%d\n", Analizza(N, M, W, L, arco_da, arco_a, capacita, R, C));

    fclose(fr);
    fclose(fw);
    return 0;
}
